### PR TITLE
Switch vulnerability scan to CLI instead of web tool [skip deploy]

### PR DESCRIPTION
**Pull request by mgwalker**
The Snyk web tool ignores `package-lock.json` and instead inspects our dependencies based on the _highest version_ of a library that our `package.json` says is acceptable.  This leads to false negatives.

The Snyk CLI tool scans what is actually installed.  Because we install explicitly from `package-lock.json` in our CircleCI config (and our deploy as well), the CLI tool will scan against the versions of libraries that we actually use.

### This pull request changes...
- no visible changes to the product
- should start seeing some failing tests

### This pull request is ready to merge when...
- [ ] This code has been reviewed by someone other than the original author

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -31,6 +31,20 @@ jobs:
...
+      - image: node:8
+    steps:
+      - checkout
+      - restore_cache:
+          key: v1-cms-hitech-apd-{{ checksum "web/package-lock.json" }}
+      - run:
+          name: vulnerability scan
+          working_directory: ~/project/web
+          command: |
+            npx snyk auth $SNYK_TOKEN
```
Comment (mgwalker) [inline .circleci/config.yml]: Right now this token is tied to my account, but the token is set in the CircleCI project settings so we can _*easily*_ switch it to someone else later with no code changes.  😄

File: bin/deploy.sh [hunk 2]
```diff
@@ -32,7 +32,7 @@ rm -rf seeds/test
 rm seeds/development.js
 rm seeds/test.js
 rm seeds/shared/delete-everything.js
-npm ci
+npm ci --only=production
```
Comment (mgwalker) [inline bin/deploy.sh]: When we deploy the API, we should only push the production dependencies.  Our dev dependencies aren't useful and there's no good argument for putting unused code into production!

**Comment by mgwalker** [comment]
🤔 Should deploys be held until vuln scans pass?  That seems like it would make sense...

File: bin/deploy.sh (hunk 2 above)
Comment (brendansudol) [reply bin/deploy.sh]: nice, agreed

Comment (brendansudol) [reply bin/deploy.sh]: also a good test that our dependencies are in the right place (dev vs. prod) in package.json

**Pull request merged**